x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/ml_inference_logic.ts

```diff
@@ -285,29 +285,22 @@ export const MLInferenceLogic = kea<
     createPipeline: () => {
       const {
         addInferencePipelineModal: { configuration, indexName },
-        isTextExpansionModelSelected,
         mlInferencePipeline, // Full pipeline definition
       } = values;
-      actions.makeCreatePipelineRequest(
-        isTextExpansionModelSelected && mlInferencePipeline && configuration.fieldMappings
-          ? {
-              indexName,
-              fieldMappings: configuration.fieldMappings,
-              pipelineDefinition: mlInferencePipeline,
-              pipelineName: configuration.pipelineName,
-            }
-          : {
-              destinationField:
-                configuration.destinationField.trim().length > 0
-                  ? configuration.destinationField
-                  : undefined,
-              indexName,
-              inferenceConfig: configuration.inferenceConfig,
-              modelId: configuration.modelID,
-              pipelineName: configuration.pipelineName,
-              sourceField: configuration.sourceField,
-            }
-      );
+
+      actions.makeCreatePipelineRequest({
+        indexName,
+        inferenceConfig: configuration.inferenceConfig,
+        fieldMappings: configuration.fieldMappings || [
+          // Temporary while we're using single fields for non-ELSER pipelines
+          {
+            sourceField: configuration.sourceField,
+            targetField: getMlInferencePrefixedFieldName(configuration.destinationField),
+          },
+        ],
+        pipelineDefinition: mlInferencePipeline || {},
```

Comment:
this should always have a value here correct? we could validate that above or just use `!` to not do the `|| {}`

Reply:
Correct. I didn't know about the `!` operator, I'll add it.